source/common/tracing/http_tracer_impl.cc

```diff
@@ -84,10 +84,21 @@ static void addGrpcRequestTags(Span& span, const Http::RequestHeaderMap& headers
 template <class T> static void addGrpcResponseTags(Span& span, const T& headers) {
   addTagIfNotNull(span, Tracing::Tags::get().GrpcStatusCode, headers.GrpcStatus());
   addTagIfNotNull(span, Tracing::Tags::get().GrpcMessage, headers.GrpcMessage());
-  // Set error tag when status is not OK.
+  // Set error tag when Grpc status code represents an error. See
+  // https://github.com/envoyproxy/envoy/issues/18877
   absl::optional<Grpc::Status::GrpcStatus> grpc_status_code = Grpc::Common::getGrpcStatus(headers);
-  if (grpc_status_code && grpc_status_code.value() != Grpc::Status::WellKnownGrpcStatus::Ok) {
-    span.setTag(Tracing::Tags::get().Error, Tracing::Tags::get().True);
+  if (grpc_status_code.has_value()) {
+    const auto& status = grpc_status_code.value();
+    if (status != Grpc::Status::WellKnownGrpcStatus::InvalidCode &&
```

Comment:
does these cover all the error status?

Reply:
Yes, detailed introduction can be found here: https://grpc.github.io/grpc/core/md_doc_statuscodes.html

Reply:
What do you think of converting to a switch, so if gRPC adds new error codes and we pick up on import, we compile fail?

Reply:
Also can we link https://grpc.github.io/grpc/core/md_doc_statuscodes.html here in the comment as well?

Reply:
Yes, I agree a switch makes more sense in this case.

Reply:
> What do you think of converting to a switch, so if gRPC adds new error codes and we pick up on import, we compile fail?

Yes, I agree a switch makes more sense in this case.

> Also can we link https://grpc.github.io/grpc/core/md_doc_statuscodes.html here in the comment as well?

Done